src/artemis/devices/oav/oav_centring.py

```diff
@@ -0,0 +1,199 @@
+
+        self.oav_parameters.load_parameters_from_file(context)
+
+        yield from bps.abs_set(self.oav.colour_mode_pv, self.oav_parameters.rgb_mode)
+        yield from bps.abs_set(
+            self.oav.acquire_period_pv, self.oav_parameters.acquire_period
+        )
+        yield from bps.abs_set(self.oav.exposure_pv, self.oav_parameters.exposure)
+        yield from bps.abs_set(self.oav.gain_pv, self.oav_parameters.gain)
+
+        # select which blur to apply to image
+        yield from bps.abs_set(
+            self.oav.preprocess_operation_pv, self.oav_parameters.preprocess
+        )
+
+        # sets length scale for blurring
+        yield from bps.abs_set(
+            self.oav.preprocess_ksize_pv, self.oav_parameters.preprocess_K_size
+        )
+
+        # Canny edge detect
+        yield from bps.abs_set(
+            self.oav.canny_lower_threshold_pv,
+            self.oav_parameters.canny_edge_lower_threshold,
+        )
+        yield from bps.abs_set(
+            self.oav.canny_upper_threshold_pv,
+            self.oav_parameters.canny_edge_upper_threshold,
+        )
+        # "Close" morphological operation
+        yield from bps.abs_set(self.oav.close_ksize_pv, self.oav_parameters.close_ksize)
+
+        # Sample detection
+        yield from bps.abs_set(
+            self.oav.sample_detection_scan_direction_pv, self.oav_parameters.direction
+        )
+        yield from bps.abs_set(
+            self.oav.sample_detection_min_tip_height_pv,
+            self.oav_parameters.minimum_height,
+        )
+
+        # Connect MXSC output to MJPG input
+        yield from self.start_mxsc(
+            self.oav_parameters.input_plugin + "." + self.oav_parameters.mxsc_input,
+            self.oav_parameters.min_callback_time,
+            self.oav_parameters.filename,
+        )
+
+        yield from bps.abs_set(
+            self.oav.input_pv, self.oav_parameters.input_plugin + ".MXSC"
+        )
+
+        # zoom is an integer value, stored as a float in a string so
+        # we may need a .0 on the end? GDA does this with
+        # zoomString = '%1.0dx' % float(zoom)
+        # which seems suspicious, we may want to think about doing this in a nicer way
+        yield from bps.abs_set(
+            self.oav_camera.zoom, f"{float(int(self.oav_parameters.zoom))}x", wait=True
+        )
+
+        if (yield from bps.rd(self.oav_backlight.control)) == "Out":
+            yield from bps.abs_set(self.oav_backlight.control, "In", wait=True)
+
+        """
+       TODO: currently can't find the backlight brightness
+        this will need to be set up after issue
+        https://github.com/DiamondLightSource/python-artemis/issues/317
+        is solved
+         brightnessToUse = self.oav_parameters.extract_dict_parameter(
+            context, "brightness"
+        )
+
+        blbrightness.moveTo(brightnessToUse)
+        """
+
+    def start_mxsc(self, input_plugin, min_callback_time, filename=None):
```

Comment:
Should: A docstring on this would be good. Also a comment about why having a filename of None is ok

Reply:
In hindsight I don't think having a filename of none is okay. This is the python file for calculating the edge waveforms, which will always be necessary. It should be a fail-state if we can't find it in the json passed from GDA imo.